### PR TITLE
Anki 23.12.1

### LINKED ISSUE #15083: Update to Anki 23.12 backend
The Anki update includes some significant changes to the FSRS algorithm. So, for best results, all reviews should be done on apps with the same codebase.

(There are no "breaking" changes, but still ...)

Now that the 23.12 release is public, the backend code of AnkiDroid should be updated as soon as possible.

### PR DESCRIPTION
**Pull request by dae**
Requires https://github.com/ankidroid/Anki-Android-Backend/pull/339

The apkg import page has been fixed. Some remaining todos:

- deck config option in export dialog
- https://github.com/ankidroid/Anki-Android/issues/15083#issuecomment-1869911711

Fixes:
- FIxes #15083 

Please squash when merging.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/AnkiServer.kt [hunk 1]
```diff
@@ -62,8 +64,12 @@ open class AnkiServer(
         val mime = getMimeFromUri(uri)
 
         if (session.method == Method.GET) {
-            val resourcePath = "web$uri"
-            val stream = this.javaClass.classLoader!!.getResourceAsStream(resourcePath)
+            val resourcePath = "backend/web$uri"
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/pages/AnkiServer.kt:67]: I didn't realize there was a draft PR open already! I see it handles the backend subdir just fine already as well, ignore me...

File: build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ buildscript {
     ext.kotlin_version = '1.9.21'
     ext.lint_version = '31.1.1'
     ext.acra_version = '5.11.3'
-    ext.ankidroid_backend_version = '0.1.32-anki23.10.1'
+    ext.ankidroid_backend_version = '0.1.34-anki23.12.1'
```
Comment (mikehardy) [inline build.gradle:12]: publishing happening right now, sometime in the next 60-90 minutes (if typical...) it'll be available at https://repo1.maven.org/maven2/io/github/david-allison-1/anki-android-backend/ and then a close/reopen here may get some more CI green?

**Comment by BrayanDSO** [comment]
I've already rebased and fixed some other stuff in [my branch](https://github.com/BrayanDSO/Anki-Android/tree/23.12), but avoided to force-push it here

File: build.gradle (hunk 2 above)
Comment (dae) [reply build.gradle:12]: Hmm, it hasn't shown up yet?

Comment (dae) [reply build.gradle:12]: Well, despite it not appearing there, the build seems to work. :man_shrugging:

**Comment by dae** [comment]
These remaining issues can be done in follow-up commits after merging:

- The filtered deck screen will need to support customizing the separate preview delays for again/hard & good buttons
- The auto advance feature will need to switch to using the settings in the current card's deck options
- The apkg export screen should expose an option to control whether deck configs are exported

**Comment by BrayanDSO** [comment]
> The filtered deck screen will need to support customizing the separate preview delays for again/hard & good buttons

If someone doesn't like temporary work like me and is very motivated, since our Filtered deck options has a lot of [deprecated API](https://github.com/ankidroid/Anki-Android/issues/5019) and some people have even called it a [rats nest](https://github.com/ankidroid/Anki-Android/issues/5019#issuecomment-427606379) because of how badly it is written, moving the page to Svelte in Anki would be the best option for the future, which [Damien has accepted](https://github.com/ankidroid/Anki-Android/issues/5019#issuecomment-1700556158), and is probably easier than handling our filtered deck options.

But if the fix is simple, it can be done as well for now, the above is just a personal recommendation.

**Comment by BrayanDSO** [comment]
note: the `Has conflict` bot didn't remove the tag for some reason

**Comment by abdnh** [comment]
> The auto advance feature will need to switch to using the settings in the current card's deck options

I'm going to look into this in two weeks (and the rest if no one picks them before that).

File: build.gradle (hunk 2 above)
Comment (mikehardy) [reply build.gradle:12]: I confirm not seeing it. That's the url I have used for confirm of publish so not blocking but not resolving yet either until figured out

**Review by mikehardy — APPROVED** [review]
These commits all look good, big thanks to @BrayanDSO for the collab and just generally keeping the backend-page-loading stuff working as well as possible as things change underneath (like packaging, this time, but seems like it is always something and you're always there with the fix!)

@dae I'd like to get the out ASAP to get the FSRS changes out, I see your comment about the rest of the items being ok to resolve post-merge, can you confirm that it is okay to release as is with this merged and the remaining items are even okay as a follow on release / future alpha?

If so I'll try to get the alpha release party train moving here

**Comment by dae** [comment]
Without the follow-up changes, users won't be able to adjust the default delays for filtered decks with rescheduling off without using the computer version, but that's probably not a big deal for a single alpha.

**Comment by mikehardy** [comment]
That sounds like an acceptable "miss" as far as features go, yes. Our alpha cadence has been about 1/week so may be more than 1 but also we can go faster if there's a PR. I'll lean towards getting the FSRS changes out given how interested folks were then. Thanks!

File: build.gradle (hunk 2 above)
Comment (mikehardy) [reply build.gradle:12]: Weirdest thing. The files are definitely there but the index listing is stale.

Proof of existence (beyond just the build working, which is sufficient proof really): 
https://repo1.maven.org/maven2/io/github/david-allison-1/anki-android-backend/0.1.34-anki23.12.1/

It is possible to see them with general "maven central" search tools though, such as 
https://central.sonatype.com/artifact/io.github.david-allison-1/anki-android-backend/versions
https://mvnrepository.com/artifact/io.github.david-allison-1/anki-android-backend

Guess I'll use the sonatype.com search next time 🤷

**Pull request merged**